File: app/src/main/java/com/project/scratchstudio/kith_andoid/CustomViews/CustomFontTextView.java
package com.project.scratchstudio.kith_andoid.CustomViews;

import android.content.Context;
import android.graphics.Typeface;
import android.util.AttributeSet;

public class CustomFontTextView extends androidx.appcompat.widget.AppCompatTextView {

    private static final String sScheme = "http://schemas.android.com/apk/res-auto";
    private static final String sAttribute = "customFont";

    public CustomFontTextView(Context context, AttributeSet attrs) {
        super(context, attrs);

        if (isInEditMode()) {
            return;
        } else {
            final String fontName = attrs.getAttributeValue(sScheme, sAttribute);

            if (fontName == null) {
                throw new IllegalArgumentException("You must provide \"" + sAttribute + "\" for your text view");
            } else {
                final Typeface customTypeface = CustomFont.fromString(fontName).asTypeface(context);
                setTypeface(customTypeface);
            }
        }
    }
}

File: app/src/main/java/com/project/scratchstudio/kith_andoid/custom_views/CustomFontTextView.kt
package com.project.scratchstudio.kith_andoid.custom_views

import android.content.Context
import android.util.AttributeSet

class CustomFontTextView(context: Context, attrs: AttributeSet) : androidx.appcompat.widget.AppCompatTextView(context, attrs) {

    init {

        if (!isInEditMode) {
            val fontName = attrs.getAttributeValue(sScheme, sAttribute)

            requireNotNull(fontName) { "You must provide \"$sAttribute\" for your text view" }
            val customTypeface = CustomFont.fromString(fontName).asTypeface(context)
            typeface = customTypeface
        }
    }

    companion object {

        private const val sScheme = "http://schemas.android.com/apk/res-auto"
        private const val sAttribute = "customFont"
    }
}